core/src/main/scala/doobie/util/meta.scala

```diff
@@ -295,6 +297,48 @@ object meta {
     implicit def ArrayTypeAsVectorMeta[A: ClassTag: TypeTag](implicit ev: Meta[Array[A]]): Meta[Vector[A]] =
       ev.xmap(a => if (a == null) null else a.toVector  , a => if (a == null) null else a.toArray)
 
+    /**
+      * Derive Meta for nullable unary product types.
+      * A - type for which instance is derived
+      * L - HList representation of type A
+      * H - type of the head of L (this is the only type in L)
+      * T - type of the tail of L (unused)
+      * @group Instances
+      */
+    implicit def unaryProductMetaNullable[A >: Null : TypeTag, L <: HList, H, T <: HList](
+      // representation (L) for type A
+      implicit gen: Generic.Aux[A, L],
+      // head (H) and tail (T) type of representation (L)
+      c: IsHCons.Aux[L, H, T],
+      // Meta instance for the head (the only element in representation)
+      hmeta: Lazy[Meta[H]],
+      // provide evidence that representation (L) and singleton hlist with
+      // the only element of type H are the same type
+      ev: =:=[H :: HNil, L],
+      ev2: Null <:< H
```

Comment:
Ah ok yeah.  I guess `Meta#nxmap` does it that way because its `A` is declared outside of `nxmap`.